scraping/get_tmdb_poster_links.py

```diff
@@ -0,0 +1,135 @@
+import requests
+from typing import Dict, Optional, List
+import os
+from urllib.parse import quote
+import json
+import time
+
+class TMDBPosterFetcher:
+    def __init__(self, api_key: Optional[str] = None):
+        """
+        Initialize the TMDB poster fetcher with an API key.
+        
+        Args:
+            api_key: TMDB API key. If not provided, will look for TMDB_API_KEY environment variable.
+        """
```

Comment:
Classes, typing, docstrings... This is great code. Meanwhile my scraper is just a random script.